crates/datafusion/src/physical_plan/string_expressions.rs

```diff
@@ -570,6 +570,29 @@ pub fn starts_with<T: StringOffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayR
     Ok(Arc::new(result) as ArrayRef)
 }
 
+/// Returns true if string ends with suffix.
+/// ends_with('alphabet', 'abet') = 't'
+pub fn ends_with<T: StringOffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
+    match args.len() {
+        2 => {
+            let string_array = downcast_string_arg!(args[0], "string", T);
+            let suffix_array = downcast_string_arg!(args[1], "suffix", T);
+            let suffix = suffix_array.value(0);
```

Comment:
sorry, my problem. this suffix_array is borrowed from starts_with? this is ok. Because the suffix is provided by the upper caller. But your change is also accepted as interesting usage if we move this method to ch_fns module.